packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -134,6 +141,13 @@ export class IncomingPayment
     return `${this.paymentPointer.url}/incoming-payments/${this.id}`
   }
 
+  public get hasConnection(): boolean {
```

Comment:
An alternative to this would be to set `connectionId` to `null` when the incoming payment transitions to `Completed`/`Expired`

Reply:
I think that would be cleaner.

Reply:
:+1: I'll change it.
My only reservation is if we don't want to lose record of the `connectionId`

Reply:
Is there any admin purpose we can think of where it would be beneficial to get the ilp credentials after the payment is `completed` or `expired`?

Reply:
Not returning the connection with a completed/expired payment is an Open Payments OpenAPI thing.
It would technically still be possible to generate ilp credentials for the incoming payment since the `connectionId` use only used for the connection url:
https://github.com/interledger/rafiki/pull/595/files#diff-b9b1a593e29f9fdc18a9826d0da73d506ccca8cca9e777165b4ac86b5a0bfdd5R72

Reply:
I know it's an OP thing. I just checked the code again and remembered that it is something completely artificial that's not even used to create the STREAM credentials. In that case, set it to `null`.